newrelic/resource_newrelic_workload.go

```diff
@@ -0,0 +1,327 @@
+		var entityGUIDs []string
+		for _, e := range workload.Entities {
+			entityGUIDs = append(entityGUIDs, *e.GUID)
+		}
+
+		createInput.EntityGUIDs = entityGUIDs
+	}
+
+	if len(workload.EntitySearchQueries) > 0 {
+		var entitySearchQueries []workloads.EntitySearchQueryInput
+		for _, q := range workload.EntitySearchQueries {
+			queryInput := workloads.EntitySearchQueryInput{
+				Name:  &q.Name,
+				Query: q.Query,
+			}
+
+			entitySearchQueries = append(entitySearchQueries, queryInput)
+		}
+
+		createInput.EntitySearchQueries = entitySearchQueries
+	}
+
+	if len(workload.ScopeAccounts.AccountIDs) > 0 {
+		var scopeAccounts workloads.ScopeAccountsInput
+		scopeAccounts.AccountIDs = []int{}
+		for _, a := range workload.ScopeAccounts.AccountIDs {
+			scopeAccounts.AccountIDs = append(scopeAccounts.AccountIDs, a)
+		}
+
+		createInput.ScopeAccountsInput = &scopeAccounts
+	}
+
+	created, err := client.Workloads.CreateWorkload(accountID, createInput)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%d:%d:%s", accountID, created.ID, created.GUID))
+	return resourceNewRelicWorkloadRead(d, meta)
+}
+
+func resourceNewRelicWorkloadRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderConfig).NewClient
+
+	ids, err := parseWorkloadIDs(d.Id())
+	if err != nil {
+		return err
+	}
+
+	workload, err := client.Workloads.GetWorkload(ids.AccountID, ids.ID)
+	if err != nil {
+		if _, ok := err.(*errors.NotFound); ok {
+			d.SetId("")
+			return nil
+		}
+
+		return err
+	}
+
+	return flattenWorkload(workload, d)
+}
+
+func resourceNewRelicWorkloadUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderConfig).NewClient
+	workload := expandWorkload(d)
+
+	log.Printf("[INFO] Updating New Relic One workload %s", d.Id())
+
+	ids, err := parseWorkloadIDs(d.Id())
+	if err != nil {
+		return err
+	}
+
+	updateInput := workloads.UpdateInput{
+		Name: &workload.Name,
+	}
+
+	if len(workload.Entities) > 0 {
+		var entityGUIDs []string
+		for _, e := range workload.Entities {
+			entityGUIDs = append(entityGUIDs, *e.GUID)
+		}
+
+		updateInput.EntityGUIDs = entityGUIDs
+	}
+
+	if len(workload.EntitySearchQueries) > 0 {
+		var entitySearchQueries []workloads.EntitySearchQueryInput
+		for _, q := range workload.EntitySearchQueries {
+			queryInput := workloads.EntitySearchQueryInput{
+				Name:  &q.Name,
+				Query: q.Query,
+			}
+
+			entitySearchQueries = append(entitySearchQueries, queryInput)
+		}
+
+		updateInput.EntitySearchQueries = entitySearchQueries
+	}
+
+	if len(workload.ScopeAccounts.AccountIDs) > 0 {
+		var scopeAccounts workloads.ScopeAccountsInput
+		scopeAccounts.AccountIDs = []int{}
+		for _, a := range workload.ScopeAccounts.AccountIDs {
+			scopeAccounts.AccountIDs = append(scopeAccounts.AccountIDs, a)
+		}
+
+		updateInput.ScopeAccountsInput = &scopeAccounts
+	}
+
+	_, err = client.Workloads.UpdateWorkload(ids.GUID, updateInput)
+	if err != nil {
+		return err
+	}
+
```

Comment:
You'll want to call another d.SetId() after you finish updating the resource.

Reply:
👍